src/main/c/truffleposix/truffleposix.c

```diff
@@ -195,3 +196,12 @@ static void copy_stat(struct stat *native_stat, struct truffleposix_stat* buffer
   buffer->gid     = native_stat->st_gid;
   buffer->uid     = native_stat->st_uid;
 }
+
+int64_t truffleposix_clock_gettime(int clock) {
+  struct timespec timespec;
+  int ret = clock_gettime((clockid_t) clock, &timespec);
```

Comment:
I think clock_gettime is only available on Linux, so you probably need to conditionally define this (the attach_function will handle it fine if the function doesn't exist).

Reply:
It's on at least macOS as well. But actually I get this failure in the gate for Linux

```
Caused by: org.truffleruby.language.control.RaiseException: /scratch/graaluser/buildslave/buildlog/ci_executor/main/lib/cext/truffleposix.so: undefined symbol: clock_gettime
```

Reply:
We need `-lrt` on Linux I think.

Reply:
I think it's available on Solaris as well. What makes you think it won't be available?

Reply:
Yeah, I think `librt` is needed on Linux.
I think OS X used to [not have](https://stackoverflow.com/questions/5167269/clock-gettime-alternative-in-mac-os-x) clock_gettime, but probably that changed.
The rdoc of `clock_gettime` also mentions some details about macOS.
Different OS support different sets of CLOCK arguments at least.

Reply:
Yes some of the clock types are platform-specific, but our core already dealt with that for the Java version of these calls.